cpp/src/strings/search/find.cu

```diff
@@ -358,17 +358,24 @@ CUDF_KERNEL void contains_warp_parallel_fn(column_device_view const d_strings,
 
   auto const str_idx  = idx / cudf::detail::warp_size;
   auto const lane_idx = idx % cudf::detail::warp_size;
+  if (lane_idx) { d_results[str_idx] = false; }
   if (d_strings.is_null(str_idx)) { return; }
   // get the string for this warp
   auto const d_str = d_strings.element<string_view>(str_idx);
-  // each thread of the warp will check just part of the string
+  // each warp processes 4 starting bytes
   auto found = false;
-  for (auto i = static_cast<size_type>(idx % cudf::detail::warp_size);
-       !found && ((i + d_target.size_bytes()) <= d_str.size_bytes());
-       i += cudf::detail::warp_size) {
+  for (auto i = lane_idx * 4; !found && ((i + d_target.size_bytes()) <= d_str.size_bytes());
```

Comment:
```suggestion
  auto constexpr bytes_per_warp = 4;
  for (auto i = lane_idx * bytes_per_warp; !found && ((i + d_target.size_bytes()) <= d_str.size_bytes());
```
Let's give it a name to avoid magic numbers. By doing so, we can probably remove the above comment as well.